File: TeamCode/src/main/java/org/hermitsocialclub/hydra/HydraConfig.java
package org.hermitsocialclub.hydra;

import com.qualcomm.hardware.bosch.BNO055IMU;
import com.qualcomm.robotcore.hardware.DcMotorController;
import com.qualcomm.robotcore.hardware.DistanceSensor;
import com.qualcomm.robotcore.hardware.HardwareMap;
import org.firstinspires.ftc.robotcore.external.navigation.VuforiaLocalizer;
import org.openftc.easyopencv.OpenCvCamera;

public abstract class HydraConfig {

    private final HardwareMap hwMap;

    public HydraConfig(HardwareMap hwMap) {
        this.hwMap = hwMap;
    }

    public abstract DcMotorController getFrontLeftWheel();

    public abstract DcMotorController getFrontRightWheel();

    public abstract DcMotorController getBackLeftWheel();

    public abstract DcMotorController getBackRightWheel();

    public abstract BNO055IMU getImu();

    public abstract DistanceSensor getDistanceSensor();

    public abstract VuforiaLocalizer.Parameters getVuforiaParameters();

    public abstract OpenCvCamera getCamera1();

    public abstract OpenCvCamera getCamera2();

}

File: TeamCode/src/main/kotlin/org/hermitsocialclub/hydra/HydraConfig.kt
package org.hermitsocialclub.hydra

import com.qualcomm.hardware.bosch.BNO055IMU
import com.qualcomm.robotcore.hardware.DcMotorController
import com.qualcomm.robotcore.hardware.DistanceSensor
import com.qualcomm.robotcore.hardware.HardwareMap
import org.firstinspires.ftc.robotcore.external.navigation.VuforiaLocalizer
import org.openftc.easyopencv.OpenCvCamera

abstract class HydraConfig(val hw: HardwareMap) {

    abstract val frontLeftWheel: DcMotorController
    abstract val frontRightWheel: DcMotorController
    abstract val backLeftWheel: DcMotorController
    abstract val backRightWheel: DcMotorController
    abstract val imu: BNO055IMU
    abstract val distanceSensor: DistanceSensor
    abstract var vuforiaParameters: VuforiaLocalizer.Parameters
    abstract val camera1: OpenCvCamera
    abstract val camera2: OpenCvCamera
}
